pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -209,18 +215,17 @@ func (spc *realStatefulPodControl) recordClaimEvent(verb string, set *apps.State
 func (spc *realStatefulPodControl) createPersistentVolumeClaims(set *apps.StatefulSet, pod *v1.Pod) error {
 	var errs []error
 	for _, claim := range getPersistentVolumeClaims(set, pod) {
-		_, err := spc.client.Core().PersistentVolumeClaims(claim.Namespace).Get(claim.Name, metav1.GetOptions{})
-		if err != nil {
-			if apierrors.IsNotFound(err) {
-				_, err := spc.client.Core().PersistentVolumeClaims(claim.Namespace).Create(&claim)
-				if err != nil {
-					errs = append(errs, fmt.Errorf("Failed to create PVC %s: %s", claim.Name, err))
-				}
-				spc.recordClaimEvent("create", set, pod, &claim, err)
-			} else {
-				errs = append(errs, fmt.Errorf("Failed to retrieve PVC %s: %s", claim.Name, err))
-				spc.recordClaimEvent("create", set, pod, &claim, err)
+		_, err := spc.pvcLister.PersistentVolumeClaims(claim.Namespace).Get(claim.Name)
+		switch {
+		case apierrors.IsNotFound(err):
+			_, err := spc.client.Core().PersistentVolumeClaims(claim.Namespace).Create(&claim)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("Failed to create PVC %s: %s", claim.Name, err))
 			}
+			spc.recordClaimEvent("create", set, pod, &claim, err)
```

Comment:
Would there ever be a situation where the create failed with AlreadyExists and the preexisting PVC is somehow "wrong" in terms of its details?